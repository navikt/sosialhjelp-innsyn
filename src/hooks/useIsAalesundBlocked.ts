import { useRouter } from "next/router";

import { useHentSoknadsStatus } from "../generated/soknads-status-controller/soknads-status-controller";
import { useHentAlleSaker } from "../generated/saks-oversikt-controller/saks-oversikt-controller";

const useIsAalesundBlocked = (): boolean => {
    const {
        query: { id },
    } = useRouter();

    const fiksDigisosId: string | undefined = !id || Array.isArray(id) ? undefined : id;
    const { data } = useHentAlleSaker();
    const { data: soknadsStatus } = useHentSoknadsStatus(fiksDigisosId!, { query: { enabled: !!fiksDigisosId } });

    if (!fiksDigisosId) {
        return !!data?.some((sak) => sak.kommunenummer === "1507");
    }

    return soknadsStatus?.kommunenummer === "1507";
};

export default useIsAalesundBlocked;
