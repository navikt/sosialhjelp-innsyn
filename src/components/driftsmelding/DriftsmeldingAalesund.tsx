import { useTranslation } from "next-i18next";
import { Alert, BodyShort, Heading } from "@navikt/ds-react";

import useIsAalesundBlocked from "../../hooks/useIsAalesundBlocked";

/**
 * Lagringstid er 15 måneder, så vennligst slett meg 1. april 2025.
 */
export const DriftsmeldingAalesund = () => {
    const { t } = useTranslation("aalesund");
    return !useIsAalesundBlocked() ? null : (
        <Alert variant="warning">
            <Heading size="small" level="2" spacing>
                {t("title")}
            </Heading>
            <BodyShort spacing>{t("message")}</BodyShort>
        </Alert>
    );
};
