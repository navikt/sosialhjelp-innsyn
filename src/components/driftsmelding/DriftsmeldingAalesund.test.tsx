import { render } from "@testing-library/react";

import useIsAalesundBlocked from "../../hooks/useIsAalesundBlocked";
import { screen } from "../../test/test-utils";

import { DriftsmeldingAalesund } from "./DriftsmeldingAalesund";

const TITLE = "Ålesund: ettersending av vedlegg blir ikke mulig";
const MESSAGE =
    "Grunnet kommunedelingen av Ålesund kommune vil det aldri bli mulig å ettersende vedlegg til søknader sendt inn før 1. januar 2024.";

jest.mock("../../hooks/useIsAalesundBlocked");

describe("DriftsmeldingAalesund", () => {
    it("renders null when not affected by Aalesund municipal division", () => {
        (useIsAalesundBlocked as jest.Mock).mockReturnValue(false);
        const { container } = render(<DriftsmeldingAalesund />);
        expect(container).toMatchSnapshot();
    });

    it("renders alert when affected by Aalesund municipal division", () => {
        (useIsAalesundBlocked as jest.Mock).mockReturnValue(true);
        const { container } = render(<DriftsmeldingAalesund />);
        expect(screen.getByText(TITLE)).toBeVisible();
        expect(screen.getByText(MESSAGE)).toBeVisible();
        expect(container).toMatchSnapshot();
    });
});
