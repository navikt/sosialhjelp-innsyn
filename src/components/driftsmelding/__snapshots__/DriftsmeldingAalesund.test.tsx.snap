// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriftsmeldingAalesund renders alert when affected by Aalesund municipal division 1`] = `
<div>
  <div
    class="navds-alert navds-alert--warning navds-alert--medium"
  >
    <svg
      aria-labelledby="title-r0"
      class="navds-alert__icon"
      fill="none"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title-r0"
      >
        Advarsel
      </title>
      <path
        clip-rule="evenodd"
        d="M12 2.25a.75.75 0 0 1 .656.387l9.527 17.25A.75.75 0 0 1 21.526 21H2.474a.75.75 0 0 1-.657-1.113l9.526-17.25A.75.75 0 0 1 12 2.25M12 8.75a.75.75 0 0 1 .75.75v4a.75.75 0 0 1-1.5 0v-4a.75.75 0 0 1 .75-.75m-1 7.75a1 1 0 1 1 2 0 1 1 0 0 1-2 0"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="navds-alert__wrapper navds-alert__wrapper--maxwidth navds-body-long navds-body-long--medium"
    >
      <h2
        class="navds-heading navds-heading--small navds-typo--spacing"
      >
        Ålesund: ettersending av vedlegg blir ikke mulig
      </h2>
      <p
        class="navds-body-short navds-body-short--medium navds-typo--spacing"
      >
        Grunnet kommunedelingen av Ålesund kommune vil det aldri bli mulig å ettersende vedlegg til søknader sendt inn før 1. januar 2024.
      </p>
    </div>
  </div>
</div>
`;

exports[`DriftsmeldingAalesund renders null when not affected by Aalesund municipal division 1`] = `<div />`;
